### PR TITLE
Resource queue remove(main only)

### PR DESCRIPTION
**Pull request by zxuejing**
Now Greenplum resource manager has 2 policies, resource queue, and resource group,
Resource manager can manage 3 kinds of resources, concurrency, CPU, and memory.

Resource group can manage concurrency more fine-grained and use control groups to control CPU,
So we decide to remove the resource queue and optimize the memory controller of the resource group.

The main modifications are as follows:
1. remove codes related to resource queue, containing src and test.
    remove gram and implementation of "create|alter|drop resource queue" and "create|alter role resource queue"
    remove view and function for resource queue, such as pg_resqueue_attributes, pg_resqueue_status
    remove catalog such as pg_resqueue, pg_resqueuecapablility,pg_resourcetype
    remove CPU controls containing sweeper progress and backoff process
    remove memory, concurrency limits, and the resource queue locks
    remove stat of the resource queue
    modify tools such as pg_dump, gpcheckcat, gpinitsystem to remove resource queue
    remove test cases related to resource queue
    modify tests to remove NOTICE: resource queue required -- using default resource queue "pg_default"
    remove some fields added for resource queue, such as the field hasResQueueLock of PortalData
2. Add a new resource manager policy RESOURCE_MANAGER_POLICY_NONE, And use it as the default value
    of Gp_resource_manager_policy.
3. If we do not use any resource manager, we still use statment_mem as the query's memory limit and use eager free to
    allocate memory to operators.
4. Add RESOURCE_QUEUE_FIXME to indicate some not-sure modifications.

For the convenience of review, I divide these modifications into three commits.

### CODE REVIEW

**Comment by soumyadeep2007** [comment]
Awesome! Thanks for tagging me on this. It will take me a few days to leave an in-depth review.

Also, we need to close any outstanding Github issues/PRs that target resource queue fixes for 7X.

**Comment by zxuejing** [comment]
> Also, we need to close any outstanding Github issues/PRs that target resource queue fixes for 7X.

Thanks, you are right! These issues or PRs about the resource queue in main branch should be closed.

File: src/include/utils/resource_manager.h [hunk 1]
```diff
@@ -15,12 +15,8 @@
 #ifndef RESOURCEMANAGER_H
 #define RESOURCEMANAGER_H
 
-#include "utils/resscheduler.h"
 #include "utils/resgroup.h"
-
-#define IsResQueueEnabled() \
-	(ResourceScheduler && \
-	 Gp_resource_manager_policy == RESOURCE_MANAGER_POLICY_QUEUE)
+#include "storage/lock.h"
```
Comment (SmartKeyerror) [inline src/include/utils/resource_manager.h:19]: Why do we need that header?

File: src/backend/postmaster/backoff.c [hunk 2]
```diff
@@ -1,1525 +0,0 @@
-/*-------------------------------------------------------------------------
```
Comment (SmartKeyerror) [inline src/backend/postmaster/backoff.c:1]: The backoff mechanism needs to be deleted too?

**Review by SmartKeyerror — COMMENTED** [review]
That's really an extraordinary work!

File: src/include/catalog/pg_proc.dat [hunk 3]
```diff
@@ -10884,13 +10871,6 @@
 { oid => 6464, descr => 'get backends of overflowed subtransaction',
    proname => 'gp_get_suboverflowed_backends', provolatile => 'v', prorettype => '_int4', proargtypes => '', prosrc => 'gp_get_suboverflowed_backends' },
 
-{ oid => 6030, descr => 'Return resource queue information',
-   proname => 'pg_resqueue_status', prorows => '1000', proretset => 't', provolatile => 'v', proparallel => 'r', prorettype => 'record', proargtypes => '', prosrc => 'pg_resqueue_status' },
-
-{ oid => 6069, descr => 'Return resource queue information',
-   proname => 'pg_resqueue_status_kv', prorows => '1000', proretset => 't', provolatile => 'v', proparallel => 'r', prorettype => 'record', proargtypes => '', prosrc => 'pg_resqueue_status_kv' },
-
```
Comment (Aegeaner) [inline src/include/catalog/pg_proc.dat:10892]: Do we need to bump the `CATALOG_VERSION_NO` for the catalog changes?

File: src/backend/postmaster/backoff.c (hunk 2 above)
Comment (zxuejing) [reply src/backend/postmaster/backoff.c:1]: backoff and sweeper-process are working together to implement CPU control.

**Comment by SmartKeyerror** [comment]
One more thing, the document under the gpdb-doc should also be updated.

**Comment by zxuejing** [comment]
> One more thing, the document under the gpdb-doc should also be updated.

thanks, after this pr is ok, we will create a story for doc team!

**Comment by adam8157** [comment]
The team decides to postpone this to the next major release, closing for now.

**Pull request closed without merging**